### PR TITLE
[Feature][Connector V2] Expose configurable options in influx db

### PR DESCRIPTION
**Pull request by FWLamb**
## Purpose of this pull request
refer to https://github.com/apache/incubator-seatunnel/issues/3391

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: docs/en/connector-v2/source/InfluxDB.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ supports query SQL and can achieve projection effect.
 | epoch              | string | no       | n             |
 | connect_timeout_ms | long   | no       | 15000         |
 | query_timeout_sec  | int    | no       | 3             |
+| common-options     |        | no       | -             |
```
Comment (EricJoy2048) [inline docs/en/connector-v2/source/InfluxDB.md]:
```suggestion
| common-options     | config       | no       | -             |
```

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/sink/InfluxDBSinkFactory.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    public OptionRule optionRule() {
+        return OptionRule.builder()
+            .required(
+                URL,
+                DATABASES,
+                KEY_MEASUREMENT,
+                KEY_TIME
+            )
+            .optional(
+                USERNAME,
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/sink/InfluxDBSinkFactory.java]: usename and password are ` bundledRequired` ?

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/sink/InfluxDBSinkFactory.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                PASSWORD,
+                CONNECT_TIMEOUT_MS,
+                KEY_TAGS,
+                BATCH_SIZE,
+                BATCH_INTERVAL_MS,
+                MAX_RETRIES,
+                RETRY_BACKOFF_MULTIPLIER_MS
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/sink/InfluxDBSinkFactory.java:63]: retry and retry_backoff_multiplier_ma are `bundledRequired ` ?

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceFactory.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder()
+            .required(
+                URL,
+                SQL,
+                DATABASES
+            )
+            .optional(
+                USERNAME,
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceFactory.java]: Same as above and please check other options.

File: docs/en/connector-v2/source/InfluxDB.md (hunk 1 above)
Comment (FWLamb) [reply docs/en/connector-v2/source/InfluxDB.md]: copy,thx.

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/sink/InfluxDBSinkFactory.java (hunk 2 above)
Comment (FWLamb) [reply seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/sink/InfluxDBSinkFactory.java]: I'll confirm,thx.

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/sink/InfluxDBSinkFactory.java (hunk 3 above)
Comment (FWLamb) [reply seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/sink/InfluxDBSinkFactory.java:63]: I'll confirm,thx.

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceFactory.java (hunk 4 above)
Comment (FWLamb) [reply seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceFactory.java]: all right.

**Comment by FWLamb** [comment]
@EricJoy2048 @531651225 PTAL.

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/sink/InfluxDBSinkFactory.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder()
+            .required(
+                URL,
+                DATABASES,
+                KEY_MEASUREMENT,
+                KEY_TIME
```
Comment (531651225) [inline seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/sink/InfluxDBSinkFactory.java]: KEY_TIME option not required，please change it.

File: docs/en/connector-v2/sink/InfluxDB.md [hunk 6]
```diff
@@ -13,20 +13,21 @@ Write data to InfluxDB.
...
-| retry_backoff_multiplier_ms | int      | no       | -                             |
-| connect_timeout_ms          | long     | no       | 15000                         |
+| name                        | type   | required | default value                |
+|-----------------------------|--------|----------|------------------------------|
+| url                         | string | yes      | -                            |
+| database                    | string | yes      |                              |
+| measurement                 | string | yes      |                              |
+| username                    | string | no       | -                            |
+| password                    | string | no       | -                            |
+| key_time                    | string | yes      | processing time              |
```
Comment (531651225) [inline docs/en/connector-v2/sink/InfluxDB.md]: the same, key_time change to not required

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/sink/InfluxDBSinkFactory.java (hunk 5 above)
Comment (FWLamb) [reply seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/sink/InfluxDBSinkFactory.java:54]: image
 
The parameter `key_time` is required in the document. Do I change it to not required?

Comment (FWLamb) [reply seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/sink/InfluxDBSinkFactory.java]: copy,thx.

File: docs/en/connector-v2/sink/InfluxDB.md (hunk 6 above)
Comment (FWLamb) [reply docs/en/connector-v2/sink/InfluxDB.md]: copy,thx.

**Review by 531651225 — DISMISSED** [review]

**Review by EricJoy2048 — DISMISSED** [review]

File: docs/en/connector-v2/sink/InfluxDB.md [hunk 7]
```diff
@@ -13,20 +13,21 @@ Write data to InfluxDB.
...
+| username                    | string | no       | -                            |
+| password                    | string | no       | -                            |
+| key_time                    | string | no       | processing time              |
+| key_tags                    | array  | no       | exclude `field` & `key_time` |
+| batch_size                  | int    | no       | 1024                         |
+| batch_interval_ms           | int    | no       | -                            |
+| max_retries                 | int    | no       | -                            |
+| retry_backoff_multiplier_ms | int    | no       | -                            |
+| connect_timeout_ms          | long   | no       | 15000                        |
+| common-options              | config | no       | -                            |
```
Comment (Hisoka-X) [inline docs/en/connector-v2/sink/InfluxDB.md:30]: Please add detail for `common-options` like other connector
image

**Review by CalvinKirs — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceFactory.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    @Override
+    public OptionRule optionRule() {
+        return OptionRule.builder()
+            .required(
+                URL,
+                SQL,
+                DATABASES
+            )
+            .bundledRequired(USERNAME, PASSWORD)
+            .bundledRequired(LOWER_BOUND, UPPER_BOUND, PARTITION_NUM, SPLIT_COLUMN)
```
Comment (ashulin) [inline seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceFactory.java]:
```suggestion
            .bundled(USERNAME, PASSWORD)
            .bundled(LOWER_BOUND, UPPER_BOUND, PARTITION_NUM, SPLIT_COLUMN)
```

**Review by ashulin — CHANGES_REQUESTED** [review]
There are code conflicts, see https://github.com/ashulin/incubator-seatunnel/blob/8ecf36af58052d28298841f4dee9707056b549a8/seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java#L193

File: seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceFactory.java (hunk 8 above)
Comment (FWLamb) [reply seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceFactory.java]: done

**Review by hailin0 — DISMISSED** [review]

**Review by Hisoka-X — APPROVED** [review]
LGTM

File: docs/en/connector-v2/sink/InfluxDB.md (hunk 7 above)
Comment (FWLamb) [reply docs/en/connector-v2/sink/InfluxDB.md:30]: done

**Review by ashulin — APPROVED** [review]
+1

**Pull request merged**